src/modules/hyprland/workspaces.cpp

```diff
@@ -863,58 +864,62 @@ std::string Workspaces::get_rewrite(std::string window_class, std::string window
   return window_rewrite_rules_.get(window_repr_key);
 }
 
-CreateWindow::CreateWindow(std::string workspace_name, WindowAddress window_address,
-                           std::string window_repr)
-    : window_(window_repr), window_address_(window_address), workspace_name_(workspace_name) {
+WorkspaceWindow::WorkspaceWindow(std::string workspace_name, WindowAddress window_address,
+                                 std::string window_repr)
+    : window_(std::move(window_repr)),
+      window_address_(std::move(window_address)),
+      workspace_name_(std::move(workspace_name)) {
   clear_addr();
   clear_workspace_name();
 }
 
-CreateWindow::CreateWindow(std::string workspace_name, WindowAddress window_address,
-                           std::string window_class, std::string window_title)
-    : window_(std::make_pair(window_class, window_title)),
-      window_address_(window_address),
-      workspace_name_(workspace_name) {
+WorkspaceWindow::WorkspaceWindow(std::string workspace_name, WindowAddress window_address,
+                                 std::string window_class, std::string window_title)
+    : window_(std::make_pair(std::move(window_class), std::move(window_title))),
+      window_address_(std::move(window_address)),
+      workspace_name_(std::move(workspace_name)) {
   clear_addr();
   clear_workspace_name();
 }
 
-CreateWindow::CreateWindow(Json::Value &client_data) {
-  window_address_ = client_data["address"].asString();
-  workspace_name_ = client_data["workspace"]["name"].asString();
-  window_ = std::make_pair(client_data["class"].asString(), client_data["title"].asString());
+WorkspaceWindow::WorkspaceWindow(Json::Value const &client_data)
+    : window_(std::make_pair(client_data["class"].asString(), client_data["title"].asString())),
+      window_address_(client_data["address"].asString()),
+      workspace_name_(client_data["workspace"]["name"].asString()) {
   clear_addr();
   clear_workspace_name();
 }
 
-std::string CreateWindow::repr(Workspaces &workspace_manager) {
+std::string WorkspaceWindow::repr(Workspaces &workspace_manager) {
   if (std::holds_alternative<Repr>(window_)) {
     return std::get<Repr>(window_);
-  } else if (std::holds_alternative<ClassAndTitle>(window_)) {
+  }
+  if (std::holds_alternative<ClassAndTitle>(window_)) {
     auto [window_class, window_title] = std::get<ClassAndTitle>(window_);
     return workspace_manager.get_rewrite(window_class, window_title);
-  } else {
-    // Unreachable
-    return "";
   }
+  // Unreachable
+  spdlog::error("WorkspaceWindow::repr: Unreachable");
+  throw std::runtime_error("WorkspaceWindow::repr: Unreachable");
```

Comment:
One thing I'm unsure of: this is unreachable code so I'm throwing here instead of returning an empty string. 

It makes more sense to me that the program throws an exception if the code is truly never supposed to reach there.

Reply:
Good one! I definitely agree a throw is better than an empty return